### ISSUE TITLE
Something weird is happening with properties in general

### ISSUE THREAD

**Issue by trackpete** [trackpete/exiletools-indexer#46]
For example:

* `properties.Dagger` should exist as a boolean, but instead it is going into `properties.Weapon.Dagger`
* why does `properties.Weapon.Armour` /etc. exist? Same with Jewelry and some other stuff. WTF?
* `properties.Jewel.The Goddess` WTF?
* and more

Note, this is probably related to https://github.com/trackpete/exiletools-indexer/issues/44

**Comment by thirdy** [comment]
Noticed this too while working on Blackmarket just now:

```
"uuid" : "1473990:b7efa8870b36ce77156d86fde90680db",
					"properties" : {
						"Weapon" : {
							"Physical DPS" : 73,
							"Dagger" : true,
							"Critical Strike Chance" : "8.89",
							"Physical Damage" : {
								"min" : 24,
								"max" : 99
							},
							"Attacks per Second" : "1.20",
							"Elemental Damage" : {
								"min" : 26,
								"max" : 57
							}
						}
					}
```

Though right now I don't use that property. I just go directly to Weapon then grab pDPS, APS etc there, given that the item in hand has a baseItemType of "Weapon". Doesn't matter right now what specific weapon type (or itemType) it is.

**Comment by thirdy** [comment]
I suspect this is caused by using "properties" as a key. It's probably confusing ES as "properties" is a keyword (or a syntax) in the 'ES mapping schema'.

Maybe changing "properties" to something like "Specific Attributes" might help or fix this.

**Comment by trackpete** [comment]
I agree `properties` is confusing with ES, but it shouldn't actually impact anything.

Looking at the properties data in the JSON array, I must've made the decision to place all properties under a Weapon as "properties.Weapon.x" - so the first question is was that the right choice, and then why isn't that always working?

Notes on troubleshooting the `properties.Weapon.Armour` issue:

```
{
  "query": {
    "filtered": {
      "filter": {
        "range": {
          "properties.Weapon.Armour": {
            "gte": 1          
          }
        }
      }
    }
  }
}
```

95 hits that have this set. Here are some md5sum's:

```
1315d4210cb1aceb35639cf4445252f4  (Saintly Chainmail detected as a Dagger)
e3fa2d2f5e5d47e4f04e779aa3cacad5 (Saintly Chainmail detected as a Dagger)
... oh crap these are all Saintly Chainmails
```

And yet:

`$gearBaseType{"Saintly Chainmail"} = "Body";`

Added some debug code to `sub.formatJSON.pl` and this came out:

`Wed Dec  9 10:18:46 2015 Matched Instilled Saintly Chainmail of the Seal to Sai`

Shit. This is a problem with our lookup table. When I created it, there were no false matches like this and I've never looked back at it.

Basically, we never get the item base type. The closest we can get is typeLine, which often contains a full name. Then we have an array of possible item base type names, and we iterate through that looking to see if the name matches.

Unfortunately this means "Sai" is matching "Saintly Chainmail."

Changing the comparison to whole word matching should fix this.

**Comment by trackpete** [comment]
As to the strange `properties.Jewel.The Goddess` ....  well, that's real:

```
{
   "w" : 1,
   "icon" : "http://webcdn.pathofexile.com/image/Art/2DItems/Jewels/unique12.png?scale=1&w=1&h=1&v=dff7876369a4b22b33affd3219adf5373",
   "support" : true,
   "typeLine" : "Cobalt Jewel",
   "corrupted" : false,
   "h" : 1,
   "sockets" : [],
   "verified" : true,
   "lockedToCharacter" : false,
   "identified" : true,
   "flavourText" : [
      "Nature provides its own solutions."
   ],
   "name" : "Survival Secrets",
   "properties" : [
      {
         "displayMode" : 0,
         "name" : "The Goddess",
         "values" : [
            [
               "1",
               0
            ]
         ]
      }
   ],
   "socketedItems" : [],
   "descrText" : "This player has DND mode enabled.",
   "frameType" : 3,
   "league" : "Darkshrine HC (IC004)",
   "explicitMods" : [
      "3 Mana Regenerated per second",
      "10% increased Elemental Damage"
   ]
}
```

Weird.

**Comment by trackpete** [comment]
Created https://github.com/trackpete/exiletools-indexer/issues/48 to fix One Handed/Two Handed properties - these should go into `properties.Weapon` instead.

**Comment by trackpete** [comment]
Changed the `IdentifyType` subroutine lookup match to use whole word boundaries:

```
      if ($data{typeLine} =~ /\b$gearbase\b/) {
```
